src/utils/DarkModeSwitch.tsx

```diff
@@ -0,0 +1,150 @@
+import React from 'react'
+import {useSpring, animated} from '@react-spring/web'
+
+export const defaultProperties = {
+  dark: {
+    circle: {
+      r: 9,
+    },
+    mask: {
+      cx: '50%',
+      cy: '23%',
+    },
+    svg: {
+      transform: 'rotate(40deg)',
+    },
+    lines: {
+      opacity: 0,
+    },
+  },
+  light: {
+    circle: {
+      r: 5,
+    },
+    mask: {
+      cx: '100%',
+      cy: '0%',
+    },
+    svg: {
+      transform: 'rotate(90deg)',
+    },
+    lines: {
+      opacity: 1,
+    },
+  },
+  springConfig: {mass: 4, tension: 250, friction: 35},
+}
+
+let REACT_TOGGLE_DARK_MODE_GLOBAL_ID = 0
+
+type SVGProps = Omit<React.HTMLAttributes<HTMLOrSVGElement>, 'onChange'>
+interface Props extends SVGProps {
+  onChange: (checked: boolean) => void
+  checked: boolean
+  style?: React.CSSProperties
+  size?: number
+  animationProperties?: typeof defaultProperties
+  moonColor?: string
+  sunColor?: string
+}
+
+export const DarkModeSwitch: React.FC<Props> = ({
+  onChange,
+  children,
+  checked = false,
+  size = 24,
+  animationProperties = defaultProperties,
+  moonColor = 'white',
+  sunColor = 'black',
+  style,
+  ...rest
+}) => {
+  const [id, setId] = React.useState(0)
```

Comment:
What's this for?

Reply:
For assigning `unique` id to the `mask` on every render, as rendering the component more than once will cause a collision with static ids.